File: bank_tech_test_ruby/lib/transaction.rb
# File: lib/transaction.rb

class Transaction

  def initialize
    @transaction = []
  end

  def get_date
    return Time.now.strftime('%d/%m/%Y')
  end

  def check_amount_valid(amount)
    fail "amount must be a positive integer" unless (amount === amount.to_i) && amount > 0 && amount.to_f % 2 == 0
  end

  def deposit(amount)
    check_amount_valid(amount)
    @transaction << get_date << amount
  end

  def withdraw(amount)
    check_amount_valid(amount)
    @transaction << get_date << - amount
  end

end

File: bank_tech_test_javascript/classes/transaction.js
class Transaction {

  constructor(transaction, date) {
    this.transaction = transaction;
    this.date = date;
  }

  getDate() {
    return new Date().toLocaleDateString()
  }

  withdraw(amount) {
    this.date = this.getDate()
    this.transaction = -amount;
  }

  deposit(amount) {
    this.date = this.getDate()
    this.transaction = amount;
  }
  
}

module.exports = Transaction;